labrad/protocol.py

```diff
@@ -51,6 +50,20 @@ def __init__(self):
 
         self.onDisconnect = util.DeferredSignal()
 
+    def set_address(self, host, port):
+        """Store host and port of remote endpoint in protocol object.
```

Comment:
It seems like the credential caching / lookup should be handled only in the BaseConnection class, not in the Protocol object (which is specific to the twisted implementation).  Why did you put it here?